codegen-server/src/test/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerInstantiatorTest.kt

```diff
@@ -233,4 +242,98 @@ class ServerInstantiatorTest {
+        val inner = model.lookup<StructureShape>("com.test#Inner")
+        val nestedStruct = model.lookup<StructureShape>("com.test#NestedStruct")
+        val sut = ServerInstantiator(codegenContext)
+
+        val project = TestWorkspace.testProject(model)
+        inner.renderWithModelBuilder(model, symbolProvider, project)
+        nestedStruct.renderWithModelBuilder(model, symbolProvider, project)
+        project.moduleFor(inner) {
+            val nestedUnion = model.lookup<UnionShape>("com.test#NestedUnion")
+            UnionGenerator(model, symbolProvider, this, nestedUnion).render()
+
+            unitTest("writable_for_maps") {
+                val data = Node.parse(
+                    """
+                    {
+                        "k1": { "map": {} },
+                        "k2": { "map": { "k3": {} } },
+                        "k3": { }
+                    }
+                    """,
+                )
+                val writables = mapOf(
+                    Pair(model.lookup<MemberShape>("com.test#Inner\$map"), writable("Some(result)")),
+                )
+                rustTemplate(
+                    """
+                    let mut result = #{Map}::new();
+                    let map = #{Map}::new();
+                    let k1 = #{Inner} { map: Some(map) };
+                    result.insert("k1".to_owned(), k1);
+                    let k3 = {
+                        let mut ret = #{Map}::new();
+                        ret.insert("k3".to_owned(), #{Inner} { map: None });
+                        ret
+                    };
+                    let k2 = #{Inner} { map: Some(k3) };
+                    result.insert("k2".to_owned(), k2);
+                    result.insert("k3".to_owned(), #{Inner} { map: None });
+                """,
+                    "Inner" to symbolProvider.toSymbol(inner),
+                    "Map" to RustType.HashMap(RustType.String, symbolProvider.toSymbol(inner).rustType()),
+                )
+                withBlock("let inner = ", ";") {
+                    sut.render(this, model.lookup("com.test#Inner"), writables, data as ObjectNode)
+                }
+                rust(
+                    """
+                    assert_eq!(inner.map().unwrap().len(), 3);
+                    assert_eq!(inner.map().unwrap().get("k1").unwrap().map.as_ref().unwrap().len(), 0);
+                    assert_eq!(inner.map().unwrap().get("k2").unwrap().map.as_ref().unwrap().len(), 1);
+                    assert_eq!(inner.map().unwrap().get("k3").unwrap().map, None);
+                    """,
+                )
+            }
+
+            unitTest("writable_preferred_over_node_data") {
+                val data = Node.parse(
+                    """
+                    {
+                        "str": "world",
+                        "num": 1
+                    }
+                    """,
+                )
+                val writables = mapOf(
```

Comment:
Again, I don't really follow why we need to create a struct to use a writable that pulls out its `.n` field for the `num` field of `NestedStruct`.

Like, shouldn't it be enough to interpolate a simple writable that e.g. writes a number to test that indeed we're using the writable?

Reply:
Right, this is because I was still falling back to the previous behavior. I've removed it now